lib/helpers/query/getEmbeddedDiscriminatorPath.js

```diff
@@ -17,7 +17,7 @@ const updatedPathsByArrayFilter = require('../update/updatedPathsByArrayFilter')
  */
 
 module.exports = function getEmbeddedDiscriminatorPath(schema, update, filter, path, options) {
-  const parts = path.split('.');
+  const parts = path.indexOf('.') === -1 ? [path] : path.split('.');
```

Comment:
Is this a performance optimization? Because i think `.split()` will always return a array with a value, even if `separator` is not found.